examples/echo-udp.rs

```diff
@@ -0,0 +1,75 @@
+//! An UDP echo server that just sends back everything that it receives.
+
+extern crate futures;
+#[macro_use]
+extern crate tokio_core;
+
+use std::{env, io};
+use std::net::SocketAddr;
+
+use futures::{Future, Poll};
+use tokio_core::net::UdpSocket;
+use tokio_core::reactor::Core;
+
+/// UDP echo server
+struct Server {
+    socket : UdpSocket,
+    buf : Vec<u8>,
+    to_send : Option<(usize, SocketAddr)>,
+}
+
+impl Server {
+    fn new(s : UdpSocket) -> Self {
+        Server {
+            socket: s,
+            to_send: None,
+            buf: vec![0u8; 1600],
+        }
+    }
+}
+
+impl Future for Server {
+    type Item = ();
+    type Error = io::Error;
+
+    fn poll(&mut self) -> Poll<(), io::Error> {
+        loop {
+            if let Some((size, peer)) = self.to_send.take() {
+                match self.socket.send_to(&self.buf[..size], &peer) {
+                    Err(e) => {
```

Comment:
I'm not super familiar with UDP, but presumably this can return "would block", right? In that case we should handle that specifically, right, and return `NotReady`?